### DISCUSSION TITLE
Having trouble with plugin on Razer Stream Controller

### DISCUSSION THREAD

**Discussion by helifella** [calibx/msfsdeck#154] [Q&A]
Hello, I am using W11 and have MSFS installed through the App Store.
I purchased a Razer Stream Controller and installed it.
I tried installing the plugin via the Loupedeck Marketplace, which threw the installation error as shown in the YouTube video. I ignored that, rebooted and changed the Profile.
My issue was that I only saw two icons on the profile:
<img width="1909" alt="Razer 2022-10-26 125041" src="https://user-images.githubusercontent.com/17753861/198151461-c0b6a622-727c-44f9-8a45-3c28c24f3903.png">
I uninstalled that and did the manual install - same issue.
Then I found if I changed the Device to CT or Live, I could see the Profile and associated control mappings applied correctly in the app, but of course changing the Live Profiles did not change the screen on the Razer hardware. So the Plugin appears to have installed as designed, but isn't working to my expectations.
The Razer is technically the same as a Loupedeck Live, and therefore anything Profile or plugin that works on the Live should work on the Razer, but something in the backend is screwing it up. I'll email Razer and Loupedeck about it too because I shouldn't be having this kind of issue- if someone writes a plugin for one, it should work on the other.
I feel ultimately this is a Razer/Loupedeck issue, but if you have a suggestion of a workaround in the interim I'd really appreciate it.
Here is the installation log, perhaps that might shed some light on it...
[2022-10-26T15-47-29-5129_LoupedeckServiceTool.log](https://github.com/calibx/msfsdeck/files/9874278/2022-10-26T15-47-29-5129_LoupedeckServiceTool.log)
Edit: I've been looking into it further and I believe the issue is the different hardware ID code of the Razer Stream Controller which is Loupedeck40. I see there are two files, DefaultProfile20win.lp4 and DefaultProfile30win.lp4, which means there also needs to be a DefaultProfile40win.lp4 for it to work on mine. It would be exactly the same as the loupedeck30 (Live) profile as the hardware is identical.
I had a look to see if I could figure out how to modify the code to achieve a solution for myself but it's a bit more advanced than I can handle - I created an issue.

**Comment by calibx** [comment]
Thank you for all the infos !
Working on it now

**Comment by calibx** [comment]
Adding the default profile for Razor and the S.
But you can already create what you want by drag/dropping controls from the right part of the UI to the center

**Comment by calibx** [comment]
Ok it is not as simple as renaming the files :(
Waiting an answer from the dev support to create it without the device

**Reply by helifella** [reply]
Thanks, I tried doing some background jiggery pokery but without success, then did as you suggested and just dragged the controls to replicate the proper setup, which is working great, including the folders which work as advertised too.
I appreciate your efforts in trying to get it working natively.

**Reply by calibx** [reply]
Can you export it ? With your profile I will have the correct header to add it by default in the plugin.

**Reply by wilddoktor** [reply]
Hi; I just purchased the Razer and came here looking for a "Razer" plugin...and found this thread. :-) Has there been any progress on creating a profile for the Razer?

**Reply by diegelmannsj** [reply]
Has anyone had success with getting this to run on the Razer Stream Controller? I've tried installing it, and (in typical Loupedeck fashion) it gives zero details as to why it failed.
<img width="1135" height="643" alt="image" src="https://github.com/user-attachments/assets/605b9749-49e9-4c58-a0b5-7ac2ed9b444a" />
<img width="1138" height="546" alt="image" src="https://github.com/user-attachments/assets/32e75350-52b1-4031-8ac7-6269b8a19dc2" />


<img width="1317" height="483" alt="image" src="https://github.com/user-attachments/assets/ff03d3b5-7e14-4c22-8bb8-52aa8de809e7" />

**Comment by calibx** [comment]
Hi, the 4.0.0 should be compatible with the Razor Controler

**Comment by wilddoktor** [comment]
Thanks! The loupedeck software (v6.0.5.21625) is asking for files with .lp4 or .lp5 extensions; should I just rename your file?

**Comment by calibx** [comment]
Hi,

Just double click on the file should trigger installation. (It works like
the the live and CT)
LP4 and 5 are for profiles not for plugin.

You can find default profile in the github plugin src.

**Reply by wilddoktor** [reply]
Ah...thank you very much! Gotta go back to work now; will try this later tonight!
